conductor/submitter.py

```diff
@@ -1445,7 +1445,15 @@ def getSelectedAvailablePackages(self):
         return selected_packages
 
     def get_package_by_id(self, package_id):
-        return self.software_packages[package_id]
+        # This can happen when the local resources files is out of sync with active
+	# packages on the back-end
+        try:
+            software_package = self.software_packages[package_id]
+        except KeyError:
+            logger.warning("Unable to locate the package {} in the list of packages.".format(package_id))
+            return None
+        
+        return software_package
```

Comment:
If it's only about a KeyError, could you not do it with `get` at the place where this function is called?

`software_packages.get(package_id)`

just a nit, so I approved anyway.

Reply:
It could be but I thought it would be helpful to have the warning message.